### ISSUE TITLE
[Question] About XCBuildKit features 

### ISSUE THREAD

**Issue by hendych** [jerrymarino/xcbuildkit#30]
Hi!
I'm in the middle of integrating Bazel into Xcode for our developers to use.
I saw this repo and I think it is amazing! Some of developers feel the Xcode stuck and not doing anything because the UI is not showing any progress.

Some developers also complaining using Bazel, Xcode doesn't show error while they typing or show any [auto-fix error](https://developer.apple.com/documentation/xcode/fixing-issues-in-your-code-as-you-type) (where we can click the error in Xcode and click button **Fix-it**)

I wonder if xcbuildkit adds that capability too.

**Comment by jerrymarino** [comment]
Hey @hendych - well congrats on the migration to Bazel, I hope to hear more about your success story in the near future 🚀 

So XCBuildKit started as an interface to enable features like this in a clean way. Mainly, a lot of devs still want to use Xcode but with Bazel. The goal here is to provide a common interface to use between all the build systems like Buck, Cmake, and Bazel. In, addition we'll provide reference implementation of the integrations. Because the rules and implementations are so divergent, the code is organized to be easy enough to adopt or tweak to different use cases of the build systems.

Yep, I suspect we can get fixits working quite simply after the next couple PRs. For fixits specifically, we might be able to easily pull this off by implement the diagnostic message - backed by serialized diagnostic and feed that back to Xcode. Once we add it to the protocol, we can know the next steps. Indexing is the main priority right now to make it better work Index While Building V2, rules_ios, BuildFarm and cloud. <- cc @thiagohmcruz

**Comment by hendych** [comment]
Hi jerry! Thank you for the detailed answer.

Turns out this plugin is what I'm looking for! Nice work guys!

**Comment by thiagohmcruz** [comment]
Yep we're focused on indexing atm but fix-its is next on the list :)
